File: app/src/main/java/com/tangem/presentation/activity/LogoActivity.java
package com.tangem.presentation.activity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;
import android.widget.TextView;

import com.tangem.wallet.BuildConfig;
import com.tangem.wallet.R;

/**
 * An example full-screen activity that shows and hides the system UI (i.e.
 * status bar and navigation/system bar) with user interaction.
 */
public class LogoActivity extends AppCompatActivity {
    public static final String TAG = LogoActivity.class.getSimpleName();

    private final Runnable mHideRunnable = new Runnable() {
        @Override
        public void run() {
            hide();
        }
    };

    private ImageView imgLogo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_logo);

        imgLogo = findViewById(R.id.imgLogo);

        // set up the user interaction to manually show or hide the system UI.
        imgLogo.setOnClickListener(view -> hide());
    }

    @SuppressLint("SetTextI18n")
    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);

        // trigger the initial hide() shortly after the activity has been created, to briefly hint to the user that UI controls are available.
        TextView AppVersion = findViewById(R.id.AppVersion);

        if (BuildConfig.DEBUG)
            AppVersion.setText("BETA v." + BuildConfig.VERSION_NAME + "\n" + "dev" + "\n" + "build " + BuildConfig.VERSION_CODE);
        else
            AppVersion.setText("BETA v." + BuildConfig.VERSION_NAME);

        if (!getIntent().getBooleanExtra("skipAutoHide", false)) {
            delayedHide(1000);
        }
    }

    private void hide() {
        Intent intent = new Intent(getBaseContext(), MainActivity.class);
        startActivity(intent);
        finish();
    }

    /**
     * Schedules a call to hide() in [delay] milliseconds, canceling any
     * previously scheduled calls.
     */
    private void delayedHide(int delayMillis) {
        //imgLogo.removeCallbacks(mHideRunnable);
        imgLogo.postDelayed(mHideRunnable, delayMillis);
    }
}

File: app/src/main/java/com/tangem/presentation/activity/LogoActivity.kt
package com.tangem.presentation.activity

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.tangem.App
import com.tangem.Constant
import com.tangem.di.Navigator
import com.tangem.wallet.BuildConfig
import com.tangem.wallet.R
import kotlinx.android.synthetic.main.activity_logo.*
import javax.inject.Inject

class LogoActivity : AppCompatActivity() {
    companion object {
        fun callingIntent(context: Context, autoHide: Boolean): Intent {
            val intent = Intent(context, LogoActivity::class.java)
            intent.putExtra(Constant.EXTRA_AUTO_HIDE, autoHide)
            return intent
        }
    }

    @Inject
    internal lateinit var navigator: Navigator

    private val hideRunnable = Runnable { this.hide() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_logo)

        App.getNavigatorComponent().inject(this)

        ivLogo.setOnClickListener { hide() }
    }

    @SuppressLint("SetTextI18n")
    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)

        if (BuildConfig.DEBUG)
            tvAppVersion.text = "BETA v." + BuildConfig.VERSION_NAME + "\n" + "dev" + "\n" + "build " + BuildConfig.VERSION_CODE
        else
            tvAppVersion.text = "BETA v." + BuildConfig.VERSION_NAME

        if (intent.getBooleanExtra(Constant.EXTRA_AUTO_HIDE, true))
            ivLogo.postDelayed(hideRunnable, Constant.MILLIS_AUTO_HIDE.toLong())
    }

    private fun hide() {
        navigator.showMain(this)
        finish()
    }

}